src/Icons/README.md

```diff
@@ -0,0 +1,112 @@
+# Symfony UX Icons
+
+Tools for embedding SVG icons in your Twig templates.
+
+```twig
+{{ ux_icon('mdi:symfony', {class: 'w-4 h-4'}) }}
+{# or #}
+<twig:UX:Icon name="mdi:check" class="w-4 h-4" />
+
+{# renders as: #}
+<svg viewBox="0 0 24 24" fill="currentColor" class="w-4 h-4"><path fill="currentColor" d="M21 7L9 19l-5.5-5.5l1.41-1.41L9 16.17L19.59 5.59z"/></svg>
+```
+
+## Installation
+
+```bash
+composer require symfony/ux-icons
+```
+
+## Icons?
+
+No icons are provided by this package but there are several ways to include and render icons. 
+
+### Local SVG Icons
+
+Add your svg icons to the `assets/icons/` directory and commit them.
+The name of the file is used as the _name_ of the icon (`name.svg` will be named `name`).
+If located in a subdirectory, the _name_ will be `sub-dir:name`.
+
+### Icons _On-Demand_
+
+[ux.symfony.com/icons](https://ux.symfony.com/icons) has a huge searchable repository of icons
+from many different sets. This package provides a way to include any icon found on this site _on-demand_.
+
+1. Visit [ux.symfony.com/icons](https://ux.symfony.com/icons) and search for an icon
+   you'd like to use. Once you find one you like, copy one of the code snippets provided.
+2. Paste the snippet into your twig template and the icon will be automatically fetched (and cached).
+3. That's it!
+
+> [!NOTE]
+> [Local SVG Icons](#local-svg-icons) of the same name will have precedence over _on-demand_ icons.
+
+#### Import Command
+
+You can import any icon from [ux.symfony.com/icons](https://ux.symfony.com/icons) to your local
+directory using the `ux:icons:import` command:
+
+ ```bash
+ bin/console ux:icons:import flowbite:user-solid # saved as `flowbite/user-solid.svg` and name is `flowbite:user-solid`
+
+ # import several at a time
+ bin/console ux:icons:import flowbite:user-solid flowbite:home-solid
+ ```
```

Comment:
> why would I want to do this versus on-demand?

We've talked a lot about how best to "lock" icons (to completely ensure you never get a different version). I really think this is the best and simplest way.

During heavy development, the on-demand system is great but I'm thinking some user's would want to lock their choices down. Simon and I talked about a `ux:icons:lock` command to look through all your templates, find icons, and import them.

Maybe we can adjust the docs to move this section lower? Perhaps use the term _lock_?

Reply:
Much clearer! I'm ok with the location anywhere. But let's talk specifically about the reason you might do this: to lock in a specific version of an icon. And that you will commit this to your repository.

Reply:
> But let's talk specifically about the reason you might do this: to lock in a specific version of an icon. And that you will commit this to your repository.

Here's some reasons but maybe @smnandre has more? I think Simon's a bit hesitant on the icons-on-demand system.

1. Like I said above, ensure the svg version is exactly what you expect (in case the version on iconify.design has changed)
2. Increase the speed of cache warming.

Reply:
Let's lock those reasons then into this section. We can iterate on it later. The important thing is that if we have 2 ways to accomplish the same thing, we at least give our users the quick pros/cons to help them decide.